docs(react-grid): add the Editing in Popup guide (#2945)
DevExpress/devextreme-reactive: packages/dx-react-grid/docs/guides/editing-in-popup.md

# React Grid - Editing in a Popup Form

> Users can edit row data in a popup edit form. This guide describes how to implement this functionality. Before you start, review the following help topics:

Reviewer: ```suggestion
This guide describes how to implement a pop-up edit form that allows users to edit row data. Before you start, review the following help topics:
```

- [Core Concepts](../../../core/docs/guides/fundamentals.md)
- [Custom Plugin Development](./custom-plugin-development.md)

The implementation consists of the following steps:

1. [Add a Custom Plugin](#add-a-custom-plugin)
1. [Handle Data Edits](#handle-data-edits)
1. [Add the Popup Edit Form](#add-the-popup-edit-form)

## Add a Custom Plugin

> For information on custom plugins, refer to the following help topic: [Custom Plugin Development](./custom-plugin-development.md). In the following code, the plugin is called `PopupEditing`, but you can rename it if required.

Reviewer: ```suggestion
For information on custom plugins, refer to the following help topic: [Custom Plugin Development](./custom-plugin-development.md). In the following code, the plugin is called `PopupEditing` (you can rename it).
```

```jsx
const PopupEditing = () => (
  <Plugin name="PopupEditing">
  </Plugin>
)
```

## Handle Data Edits

To handle data edits, add the following functions:

- `processValueChange` - processes user input
- `applyChanges` - applies data edits
- `cancelChanges` - cancels data edits

> Implement the functions in the `popupEditing` template and pass them to the `Popup` component along with the row being edited. Popup form elements will use these functions later.

Reviewer: ```suggestion
Implement the functions in the `popupEditing` template and pass these functions and the row being edited to the `Popup` component. Pop-up form elements use these functions later.
```

              change: createRowChange(editedRow, value, name),
            };
            if (isNew) {
              changeAddedRow(changeArgs);
            } else {
              changeRow(changeArgs);
            }
          };
          const rowIds = isNew ? [0] : editingRowIds;
          const applyChanges = () => {
            if (isNew) {
              commitAddedRows({ rowIds });
            } else {
              stopEditRows({ rowIds });
              commitChangedRows({ rowIds });
            }
          };
          const cancelChanges = () => {
            if (isNew) {
              cancelAddedRows({ rowIds });
            } else {
              stopEditRows({ rowIds });
              cancelChangedRows({ rowIds });
            }
          };

          const open = editingRowIds.length > 0 || isNew;
          return (
            <Popup
              open={open}
              row={editedRow}
              onChange={processValueChange}
              onApplyChanges={applyChanges}
              onCancelChanges={cancelChanges}
            />
          );
        }}
      </TemplateConnector>
    </Template>
    {/* ... */}
  </Plugin>
```

## Add the Popup Edit Form

> Implement a component that renders a popup form with multiple text fields and Save and Cancel buttons. Each text field should execute the `processValueChange` function when the `onChange` event is raised. The buttons should execute the `applyChanges` and `cancelChanges` functions.

Reviewer 2: ```suggestion
Implement the `Popup` component from the previous step. This component renders popup form elements: a title, multiple text fields in a grid, Save and Cancel buttons. When the `onChange` event is raised, each text field executes the `onChange` function passed to the `Popup` (`processValueChange` in the previous step). Similarly, the buttons execute the `Popup`'s `onApplyChanges` and `onCancelChanges` functions (`applyChanges` and `cancelChanges`). All popup form elements are components from the Material-UI framework.
```

> Implement the `Popup` component from the previous step. This component renders popup form elements: a title, multiple text fields in a grid, Save and Cancel buttons. When the `onChange` event is raised, each text field executes the `onChange` function passed to the `Popup` (`processValueChange` in the previous step). Similarly, the buttons execute the `Popup`'s `onApplyChanges` and `onCancelChanges` functions (`applyChanges` and `cancelChanges`). All popup form elements are components from the Material-UI framework.

Reviewer: ```suggestion
Implement the `Popup` component from the previous step. This component renders pop-up form elements: a title, multiple text fields in a grid, and Save and Cancel buttons. When the `onChange` event is raised, each text field executes the `onChange` function passed to the `Popup` (`processValueChange` in the previous step). Similarly, the buttons execute the `Popup`'s `onApplyChanges` and `onCancelChanges` functions (`applyChanges` and `cancelChanges`). All pop-up form elements are components from the Material-UI framework.
```
